### PR TITLE
Zarr to wkw converter

### PR DESCRIPTION
**Pull request by erjel**
While advertised [differently](https://webknossos.org/features) there seems no export from `zarr` arrays into the `wkw` format. Trivial approaches, such as loading the full data into memory and writing in bulk, become quickly prohibitive for large datasets which easily exhaust RAM resources of the the used system.

Therefore, I created the `convert_zarr.py` routine for the conversion of segmentation layers. The file borrows large parts from the `convert_raw.py` and probably can be extended to a base class for the chunk-wise conversion of numpy-like file formats such as `.npy`, `.hdf5`, `n5`, and others.

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [ ] Updated Changelog

### CODE REVIEW

File: wkcuber/wkcuber/convert_zarr.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+def convert_zarr(
+    source_zarr_path: Path,
+    target_path: Path,
+    layer_name: str,
+    scale: Optional[Tuple[float, float, float]] = (1.0, 1.0, 1.0),
+    flip_axes: Optional[Union[int, Tuple[int, ...]]] = None,
+    compress: bool = True,
+    file_len: int = DEFAULT_WKW_FILE_LEN,
+    executor_args: Optional[argparse.Namespace] = None,
+) -> MagView:
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_zarr.py:135]:
```suggestion
def convert_zarr(
    source_zarr_path: Path,
    target_path: Path,
    layer_name: str,
    is_segmentation_layer: bool = False,
    scale: Optional[Tuple[float, float, float]] = (1.0, 1.0, 1.0),
    flip_axes: Optional[Union[int, Tuple[int, ...]]] = None,
    compress: bool = True,
    file_len: int = DEFAULT_WKW_FILE_LEN,
    executor_args: Optional[argparse.Namespace] = None,
) -> MagView:
```

File: wkcuber/wkcuber/convert_zarr.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    f = zarr.open(source_zarr_path, "r")
+    input_dtype = f.dtype
+    shape = f.shape
+
+    if scale is None:
+        scale = 1.0, 1.0, 1.0
+    wk_ds = Dataset(target_path, scale=scale, exist_ok=True)
+    wk_layer = wk_ds.get_or_add_layer(
+        layer_name,
+        "segmentation",
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_zarr.py:147]:
```suggestion
        "segmentation" if is_segmentation_layer else "color",
```

File: wkcuber/wkcuber/convert_zarr.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                    target_mag_view=wk_mag,
+                    flip_axes=flip_axes,
+                ),
+                wk_layer.bounding_box.chunk(
+                    chunk_size=(DEFAULT_WKW_VOXELS_PER_BLOCK * file_len,) * 3
+                ),
+            )
+        )
+
+    is_segmentation_layer = True
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_zarr.py:171]:
```suggestion
```

File: wkcuber/wkcuber/convert_zarr.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+    add_scale_flag(parser, required=False)
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the cubed layer (color or segmentation).",
+        default="color",
+    )
+
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_zarr.py:61]:
```suggestion

    parser.add_argument(
        "--is_segmentation_layer",
        "-sl",
        help="Set whether this is a segmentation layer, defaulting to a color layer if nothing is set.",
        default=False,
        action="store_true",
    )

```

File: wkcuber/wkcuber/convert_zarr.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    mag_view = convert_zarr(
+        source_path,
+        args.target_path,
+        args.layer_name,
+        args.scale,
+        args.flip_axes,
+        not args.no_compress,
+        executor_args=executor_args,
+    )
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_zarr.py:199]:
```suggestion
    mag_view = convert_zarr(
        source_path,
        args.target_path,
        layer_name=args.layer_name,
        is_segmentation_layer=args.is_segmentation_layer,
        scale=args.scale,
        flip_axes=args.flip_axes,
        compress=not args.no_compress,
        executor_args=executor_args,
    )
```

**Review by jstriebel — COMMENTED** [review]
Thanks a lot for your contribution @erjel! :rocket: 

This mostly looks good to me. It would be great if you could make it an option if the layer should be of type color or category. I added some code-suggestions in the comments. Then this should be ready to be merged.

**Comment by erjel** [comment]
@jstriebel: I added the requested changes.

**Review by jstriebel — APPROVED** [review]
Awesome, thanks a lot :+1:

**Pull request merged**